video/video.ino

```diff
@@ -218,13 +223,43 @@ void setConsoleMode(bool mode) {
 
 // Switch to terminal mode
 //
-void switchTerminalMode() {
+void switchTerminalMode(bool enable) {
+	if (terminalMode == enable) {
+		return;
+	}
+	exitTerminal = false;
 	cls(true);
 	canvas.reset();
-	Terminal.begin(_VGAController.get());	
-	Terminal.connectSerialPort(VDPSerial);
-	Terminal.enableCursor(true);
-	terminalMode = true;
+	if (enable) {
+		Terminal = std::unique_ptr<fabgl::Terminal>(new fabgl::Terminal());
+		Terminal->begin(_VGAController.get());	
+		Terminal->connectSerialPort(VDPSerial);
+		Terminal->enableCursor(true);
+		// onVirtualKey is triggered whenever a key is pressed or released
+		Terminal->onVirtualKeyItem = [&](VirtualKeyItem * vkItem) {
+			if (vkItem->vk == VirtualKey::VK_F12) {
+				if (vkItem->CTRL && (vkItem->LALT || vkItem->RALT)) {
+					// CTRL + ALT + F12: emergency exit terminal mode
+					exitTerminal = true;

```

Comment:
this key combo of ctrl+alt+f12 is kinda an arbitrary choice.  it's copied from the fab-gl AnsiTerminal.ino example

_arguably_ it's too simple a combo, as it is potentially possible that this combo may be used by an application.  this code _could_ check for more modifiers.

I'd be interested in hearing others thoughts on this